File: SharePoint/CSOM/sharepoint-2013-client-object-model-rest/materials/9-sharepoint-2013-client-object-model-rest-m9-services-exercise-files/Demos/CS/ManagedCodeDemo/ManagedCodeDemo/Form1.cs
using Microsoft.SharePoint.Client;
using Microsoft.SharePoint.Client.Search.Query;
using Microsoft.SharePoint.Client.Taxonomy;
using Microsoft.SharePoint.Client.UserProfiles;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Net;
using System.Security.Principal;
using System.Text;
using System.Threading.Tasks;
using System.Web.Script.Serialization;
using System.Windows.Forms;
using System.Xml.Linq;

namespace ManagedCodeDemo
{
    public partial class Form1 : System.Windows.Forms.Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void ProfileButton_Click(object sender, EventArgs e)
        {
            var userName = WindowsIdentity.GetCurrent().Name;
            var proxy = new UserProfileProxy.UserProfileService();
            proxy.UseDefaultCredentials = true;
            proxy.Url = "http://localhost/sites/demo/_vti_bin/userprofileservice.asmx";
            var props = proxy.GetUserProfileByName(userName);
            ResultsListBox.Items.Add("User profile properties for " + userName);
            for (int i = 0; i < props.Length; i++)
            {
                var item = props[i];
                var value = item.Values.Length > 0 ? item.Values[0].Value : null;
                if (value != null)
                {
                    ResultsListBox.Items.Add(item.Name + ": " + value);
                }
            }
        }

        private void ListDataButton_Click(object sender, EventArgs e)
        {
            var serviceUrl = "http://localhost/sites/demo/_vti_bin/ListData.svc";
            var context = new DemoProxy.DemoDataContext(new Uri(serviceUrl));
            context.Credentials = System.Net.CredentialCache.DefaultCredentials;

            var category = "Beverages";
            var products = from product in context.Products
                           where product.UnitsInStock > 0 &&
                            product.Category.Title == category
                           select product;

            ResultsListBox.Items.Add(category);
            foreach (var product in products)
            {
                ResultsListBox.Items.Add(product.Title);
            }

        }

    }
}

File: SharePoint/CSOM/CSOM slides/materials/exercise9/Demos/VB/ManagedCodeDemo/ManagedCodeDemo/Form1.vb
Imports Microsoft.SharePoint.Client
Imports System.Net
Imports System.Web.Script.Serialization
Imports Microsoft.SharePoint.Client.Taxonomy
Imports Microsoft.SharePoint.Client.UserProfiles
Imports Microsoft.SharePoint.Client.Search.Query
Imports System.Security.Principal

Public Class Form1

    Private Sub ProfileButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles ProfileButton.Click
        Dim userName = WindowsIdentity.GetCurrent().Name
        Dim proxy = New UserProfileProxy.UserProfileService()
        proxy.UseDefaultCredentials = True
        proxy.Url = "http://localhost/sites/demo/_vti_bin/userprofileservice.asmx"
        Dim props = proxy.GetUserProfileByName(userName)
        ResultsListBox.Items.Add("User profile properties for " & userName)
        For i As Integer = 0 To props.Length - 1
            Dim item = props(i)
            Dim value = If(item.Values.Length > 0, item.Values(0).Value, Nothing)
            If value IsNot Nothing Then
                ResultsListBox.Items.Add(item.Name & ": " & value.ToString())
            End If
        Next i
    End Sub

    Private Sub ListDataButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles ListDataButton.Click
        Dim serviceUrl = "http://localhost/sites/demo/_vti_bin/ListData.svc"
        Dim context = New DemoProxy.DemoDataContext(New Uri(serviceUrl))
        context.Credentials = System.Net.CredentialCache.DefaultCredentials

        Dim category = "Beverages"
        Dim products = From product In context.Products
                       Where product.UnitsInStock > 0 AndAlso product.Category.Title = category
                       Select product

        ResultsListBox.Items.Add(category)
        For Each product In products
            ResultsListBox.Items.Add(product.Title)
        Next product

    End Sub

End Class
